<?xml version="1.0"?>

<robot xmlns:xacro="http://ros.org/wiki/xacro">

  <xacro:macro name="arm" params="parent x y z">

    <joint name="joint_fix" type="fixed">
      <origin xyz="${x} ${y} ${z}"/>
      <parent link="${parent}"/>
      <child link="base_link"/>
    </joint>

    <link name="base_link">
      <collision>
        <origin xyz="0 0 0.2" rpy="0 0 0"/>
        <geometry>
          <cylinder length="0.4" radius="0.1"/>
        </geometry>
      </collision>

      <visual>
        <origin xyz="0 0 0.2" rpy="0 0 0"/>
        <geometry>
          <cylinder length="0.4" radius="0.1"/>
        </geometry>
      </visual>

      <inertial>
        <origin xyz="0 0 0.2" rpy="0 0 0"/>
        <mass value="100"/>
        <inertia
          ixx="1.0" ixy="0.0" ixz="0.0"
          iyy="1.0" iyz="0.0"
          izz="1.0"/>
      </inertial>
    </link>

    <joint name="rotation1" type="revolute">
      <parent link="base_link"/>
      <child link="arm1"/>
      <origin xyz="0 0 0.4" rpy="0 0 0"/>
      <axis xyz="0 0 1"/>
      <limit effort="1000" velocity="1" lower="-2.6" upper="2.6"/>
    </joint>

    <link name="arm1">
      <collision>
        <origin xyz="0 -0.5 0.05" rpy="0 0 0"/>
        <geometry>
          <box size="0.1 0.9 0.1"/>
        </geometry>
      </collision>

      <visual>
        <origin xyz="0 -0.5 0.05" rpy="0 0 0"/>
        <geometry>
          <box size="0.1 0.9 0.1"/>
        </geometry>
      </visual>

      <inertial>
        <origin xyz="0 0 0.05" rpy="0 0 0"/>
        <mass value="50"/>
        <inertia
          ixx="1" ixy="0.0" ixz="0.0"
          iyy="1" iyz="0.0"
          izz="1"/>
      </inertial>
    </link>

    <joint name="rotation2" type="revolute">
      <parent link="arm1"/>
      <child link="arm2"/>
      <origin xyz="0 -1 0.1" rpy="0 0 0"/>
      <axis xyz="0 0 1"/>
      <limit effort="1000" velocity="1" lower="-3.00" upper="3.00"/>
    </joint>

    <link name="arm2">
      <collision>
        <origin xyz="0 -0.4 0.05" rpy="0 0 0"/>
        <geometry>
          <box size="0.1 0.9 0.1"/>
        </geometry>
      </collision>

      <visual>
        <origin xyz="0 -0.4 0.05" rpy="0 0 0"/>
        <geometry>
          <box size="0.1 0.9 0.1"/>
        </geometry>
      </visual>

      <inertial>
        <origin xyz="0 0 0.05" rpy="0 0 0"/>
        <mass value="10"/>
        <inertia
          ixx="1" ixy="0.0" ixz="0.0"
          iyy="1" iyz="0.0"
          izz="1"/>
      </inertial>
    </link>
    
  </xacro:macro>

</robot>
